pkg/kube/statuswait.go

```diff
@@ -0,0 +1,180 @@
+/*
```

Comment:
If we are moving this logic, perhaps put it into its own (internal) package?

Reply:
I moved the statusreaders to their own package in internal, but I think it would be a good idea to keep the waiters in kube, for this PR. We'd have to change the Waiter interface to not accept a kube.ResourceList to avoid cyclic imports. This PR is already XXL, I'd like to avoid more changes.